src/ray/util/logging.cc

```diff
@@ -196,15 +286,57 @@ void RayLog::StartRayLog(const std::string &app_name, RayLogLevel severity_thres
         app_name_without_path = app_file_name;
       }
     }
+#ifdef RAY_USE_GLOG
     app_name_without_path += ".";
     google::SetLogFilenameExtension(app_name_without_path.c_str());
     google::SetLogDestination(level, dir_ends_with_slash.c_str());
     FLAGS_stop_logging_if_full_disk = true;
+#else
+#ifdef _WIN32
+    int pid = _getpid();
+#else
+    pid_t pid = getpid();
+#endif
+    // Reset log pattern and level and we assume a log file can be rotated with
+    // 10 files in max size 512M.
+    // Format pattern is 2020-08-21 17:00:00,000 I 100 1001 msg.
+    // %L is loglevel, %P is process id, %t for thread id.
+    spdlog::set_pattern("[%Y-%m-%d %H:%M:%S,%e %L %P %t] %v");
+    spdlog::set_level(static_cast<spdlog::level::level_enum>(severity_threshold_));
+    // Sink all log stuff to default file logger we defined here. We may need
+    // multiple sinks for different files or loglevel.
+    auto file_logger = spdlog::get("ray_log_sink");
+    if (!file_logger) {
+      file_logger =
+          spdlog::rotating_logger_mt("ray_log_sink",
+                                     dir_ends_with_slash + app_name_without_path + "_" +
+                                         std::to_string(pid) + ".log",
+                                     1 << 29, 10);
```

Comment:
Can we make this part configurable?

Reply:
I'll extract theses parameters from env vars.

> Can we make this part configurable?